tests/client_decode.rs

```diff
@@ -0,0 +1,67 @@
+mod client_decode {
```

Comment:
This doesn't need a surrounding `mod`; it wraps tests out of the box already.

Reply:
@yoshuawuyts I know these aren't _necessary_ but I like that I can type `cargo test client_decode`

Reply:
(without the extra `mod client_decode {` line, it doesn't seem work to type `cargo test client_decode`)